ckanext/validation/logic.py

```diff
@@ -448,9 +450,14 @@ def resource_create(up_func, context, data_dict):
         {'id': package_id})
 
     t.check_access('resource_create', context, data_dict)
-
-    for plugin in plugins.PluginImplementations(plugins.IResourceController):
-        plugin.before_create(context, data_dict)
+    
+    # Check if CKAN version is min 2.10
+    if ckan_2_10:
+        for plugin in plugins.PluginImplementations(plugins.IResourceController):
```

Comment:
Alternatively: Conditionally retrieve the function object before starting the loop. Something like:

    plugin_function = plugin.before_resource_create if ckan_2_10 else plugin.before_create
    for plugin in plugins.PluginImplementation(plugins.IResourceController):
        plugin_function(context, data_dict)